gh-119287: clarify doc on BaseExceptionGroup.derive and link to it from contextlib.suppress (#119657)
python/cpython: Doc/library/contextlib.rst

   If the code within the :keyword:`!with` block raises a
   :exc:`BaseExceptionGroup`, suppressed exceptions are removed from the

>    group.  If any exceptions in the group are not suppressed, a group containing them
>    is re-raised (see also :meth:`~BaseExceptionGroup.derive`).

Author: ```suggestion
   group.  If any exceptions in the group are not suppressed, a group containing
   them is re-raised (see also :meth:`~BaseExceptionGroup.derive`).
```

>    them is re-raised (see also :meth:`~BaseExceptionGroup.derive`).

Reviewer: ```suggestion
   them is re-raised using the group's :meth:`~BaseExceptionGroup.derive` method.
```

Author: That wording is not quite correct - derive is not what re-raises it, it plays a part in the process.

Author: Also, it may not use the BaseExceptionGroup.derive method, if you overwrite it in the subclass (the point here is that you should consider overriding it).

Reviewer: > Also, it may not use the BaseExceptionGroup.derive method, if you overwrite it in the subclass

I understood that, but it will render as `derive()` in the docs rather than `BaseException.derive()` -- the `~` ensures that.

Maybe something like this?

```suggestion
   them is created using the old group's :meth:`~BaseExceptionGroup.derive`
   method, and the new group is then re-raised.
```

   If the code within the :keyword:`!with` block raises a
   :exc:`BaseExceptionGroup`, suppressed exceptions are removed from the

>    group.  If any exceptions in the group are not suppressed, a group containing
>    them is created using the old group's :meth:`~BaseExceptionGroup.derive`
>    method, and the new group is then re-raised.

Author: ```suggestion
   group.  Any exceptions of the group which are not suppressed are re-raised in
   a new group which is created using the original group's :meth:`~BaseExceptionGroup.derive`
   method.
```

How about that?

Reviewer: Nice!

Reviewer: Or "created using" rather than "created by"?
